### DISCUSSION TITLE
Getting "Target not mounted" in Multi page react application

### DISCUSSION THREAD

**Discussion by Pbaswaraj1** [gilbarbara/react-joyride#1056] [Q&A]
Hi, 

I am trying to use react-guide tour for multipage  react router application, getting this error "Target not mounted". is there any solution to make it work in the multiple page react application and i am using react router for routing.

Thanks in Advance!!

**Comment by gilbarbara** [comment]
Hey @Pbaswaraj1 

Have you checked the MultiRouter [example](https://codesandbox.io/p/devbox/github/gilbarbara/react-joyride-demo/tree/main/?file=%2Fsrc%2FMultiRoute%2Findex.tsx) in the demo?

**Comment by Pbaswaraj1** [comment]
Thanks for responding @gilbarbara. 
Yes, I did. In my case the whole route URL is changing and the page is re-rendering. this is the reason i am getting above error.

**Reply by gilbarbara** [reply]
Are you stopping the tour before changing the route and starting it again when the new page is mounted?

**Reply by Pbaswaraj1** [reply]
Nope, i am not doing that

**Reply by gilbarbara** [reply]
Check this [comment](https://github.com/gilbarbara/react-joyride/discussions/1049#discussioncomment-10025868) for further information.

**Reply by Pbaswaraj1** [reply]
Sure, thanks for sharing!!
